base9/Settings.cpp

```diff
@@ -509,6 +511,24 @@ template<typename T> T Settings::getOrDie (Node & n, string && f)
     }
 }
 
+template<typename T> T Settings::getOrRequest (Node & n, string && f)
+{
+    if (n[f])
+    {
+        return n[f].as<T> ();
+    }
+    else
+    {
+        string input = "";
+        cout << "Please enter your desired setting for field: '" + f + "':\n";
+        getline(cin, input);
+        cout << "Field '" + f + "' set to: " << input << endl;
+        YAML::Node node;
```

Comment:
Yeah that would be a better way to handle it, agreed.  The issue was just that I was unfamiliar with namespaces, so I need to take a minute or two to read about how they work.  It looks like the changes necessary though are relatively simple, reading your comments.